src/utils/fs.rs

```diff
@@ -87,24 +87,40 @@ pub fn get_all_files<A: AsRef<Path>>(
         })
     });
 
-    let types: Option<&[LanguageType]> = config.types.as_ref().map(|v| &**v);
```

Comment:
Thank you for your PR, thinking on this I'm against this change. The current changes adds too much duplication to be worth a potential performance increase. However reviewing this I did figure out a way to refactor the condition that would result in less duplication and remove checking the option. I've included how it would look below. 

```rust
let types: &[LanguageType] = config.types.as_ref().map(|v| &**v).unwrap_or(&[]);
let types_is_empty = types.is_empty();

rx.into_iter()
// …
 .filter(|(_, l)| types_is_empty || types.contains(l))
```

Reply:
@XAMPPRocky but doesn't that still requires it to go through the `filter`? I don't think the compiler get to optimize that out, I also don't know how to check.

Reply:
Oh sorry, I actually didn't see that you didn't include that in the second branch. I'd still like to avoid having code so similar duplicated. I would accept an implementation that used a macro, though if it can be avoided, that would be preferred.

Reply:
I did try removing that line but the type have issues, and also I don't really want to use a `dyn`. What do you think of this?
```rust
let rx_iter = rx
    .into_iter()
    .par_bridge()
    .filter_map(|e| LanguageType::from_path(e.path(), &config).map(|l| (e, l)));

let process = |(entry, language): (DirEntry, LanguageType)| {
    let result = language.parse(entry.into_path(), &config);
    let mut lock = languages.lock();
    let entry = lock.entry(language).or_insert_with(Language::new);
    match result {
        Ok(stats) => entry.add_stat(stats),
        Err((error, path)) => {
            entry.mark_inaccurate();
            error!("Error reading {}:\n{}", path.display(), error);
        }
    }
};

if let Some(types) = config.types.as_ref().map(|v| &**v) {
    rx_iter.filter(|(_, l)| types.contains(l)).for_each(process)
} else {
    rx_iter.for_each(process)
}
```

Reply:
That looks good to me! I think that's about as DRY you can get without using dyn or macros.

Reply:
Okay, I added this. How should we test for regression?

Reply:
Feel free to.